### PR TITLE
Created derive macro for BlockData trait

### PR DESCRIPTION
**Pull request by kallekankaanpaa**
Created derive macro for BlockData trait.

### CODE REVIEW

File: crates/macros/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+}
+
+fn impl_block_data(ast: &DeriveInput) -> TokenStream {
+    let name = &ast.ident;
+    let field;
+
+    if let Data::Struct(data) = &ast.data {
+        if let Fields::Named(named_fields) = &data.fields {
+            if named_fields.named.len() != 1 {
+                panic!("Can't derive BlockData for struct with more than 1 named field")
```
Comment (Defman) [inline crates/macros/src/lib.rs]: Errors in procmacro should be handled with or similar
https://docs.rs/syn/1.0.60/syn/struct.Error.html#method.new

**Review by Defman — CHANGES_REQUESTED** [review]
Looks good, error handling could be done a bit differently.

File: crates/macros/src/lib.rs (hunk 1 above)
Comment (kallekankaanpaa) [reply crates/macros/src/lib.rs]: Thanks for pointing that out!

File: crates/macros/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    impl_block_data(&ast).unwrap_or_else(|err| err.to_compile_error().into())
+}
+
+fn impl_block_data(ast: &DeriveInput) -> Result<TokenStream> {
+    let name = &ast.ident;
+    let field;
+
+    if let Data::Struct(data) = &ast.data {
+        if let Fields::Named(named_fields) = &data.fields {
+            if named_fields.named.len() != 1 {
```
Comment (caelunshun) [inline crates/macros/src/lib.rs]: I think we need support for multiple fields, since most of Spigot's `BlockData`s do provide multiple properties.

Comment (kallekankaanpaa) [reply crates/macros/src/lib.rs]: Updated to support multiple fields

**Review by caelunshun — APPROVED** [review]
Looks good, thanks!

**Pull request merged**